File: springapp/src/main/java/com/examly/springapp/model/RazorpayPayment.java
package com.examly.springapp.model;

import java.time.LocalDateTime;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.OneToOne;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

@Entity
@Getter
@Setter
@ToString
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class RazorpayPayment {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    Long id;

    String paymentId;
    String orderId;
    String signature;

    String currency;
    String status;
    Long amount;

    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;

    @OneToOne
    @JoinColumn(name = "appointment_id", unique = true)
    private Appointment appointment;

}

File: angularapp/src/app/models/razorpay-payment.ts
import { Appointment } from "./appointment.model";

export interface RazorpayPayment {
    id?: number;
    paymentId?: string;
    orderId?: string;
    signature?: string;
    currency: string;
    status?: string;
    amount: number;
    createdAt?: string;
    updatedAt?: string;
    appointment: Appointment;
  }
  
